src/SPCPAuthClient.class.ts

```diff
@@ -316,18 +343,18 @@ class SPCPAuthClient {
 // Functions for extracting attributes from Artifact Response
 SPCPAuthClient.extract = {
   CORPPASS: ([element]) => {
-    const cpXMLBase64 = xpath.select('string(./*[local-name(.)=\'AttributeValue\'])', element)
+    const cpXMLBase64 = xpath.select('string(./*[local-name(.)=\'AttributeValue\'])', element) as unknown as string
     return xml2json(base64.decode(cpXMLBase64))
   },
   SINGPASS: attributeElements => attributeElements.reduce(
     (attributes, element) => {
-      const key = xpath.select('string(./@Name)', element)
+      const key = xpath.select('string(./@Name)', element) as unknown as string
```

Comment:
I _think_ what's happening here is `key` is really a `string[]`, but it gets converted to a comma-delimited string when it is used as an object key. this casting is the closest way I could think of to represent this.

Reply:
I think we expect a single value for each element passed, so how about...
```suggestion
      const key = xpath.select('string(./@Name)', element, true)!.toString()
```